app/src/org/commcare/preferences/DeveloperPreferences.java

```diff
@@ -378,4 +394,30 @@ public static boolean shouldShowUpdateOptionsSetting() {
                 CommCarePreferences.YES) || BuildConfig.DEBUG;
     }
 
+    private void hideDangerousDeveloperPrefsIfNeeded() {
+        if (!GlobalPrivilegesManager.isAdvancedSettingsAccessEnabled() && !BuildConfig.DEBUG) {
+            // Dangerous privileges should not be showing
+            PreferenceScreen prefScreen = getPreferenceScreen();
```

Comment:
We could iterate over this with
prefScreen.getPreferenceCount -> prefScreen.getPReference(cout)

and just hide anything that doesn't show up in the whitelisted keys, which would be a bit less verbose (and less error prone) than building both lists and diffint